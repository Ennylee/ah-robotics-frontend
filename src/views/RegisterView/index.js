import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import RegisterComponent from '../../components/Register';
import { registerUser } from '../../redux/actions/RegisterActions/actions';

class RegisterView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      username: '',
      password: '',
      confirmPassword: '',
      errors: {},
    };
    // this.onChange = this.onChange.bind(this);
    // this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillReceiveProps(nextprops) {
    if (nextprops.register.success === true) {
      this.props.history.push('/login');
    }
  }

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = (e) => {
    const { registerUser: registerAction } = this.props;
    e.preventDefault();
    const registerData = {
      email: this.state.email,
      username: this.state.username,
      password: this.state.password,
    };
    registerAction(registerData);
  }

  render() {
    const loading = this.props.register.isRegistering;
    return (
      <RegisterComponent
        onChange={this.onChange}
        handleSubmit={this.handleSubmit}
        registerSuccess={this.props.register}
        errors={this.props.register.errors ? this.props.register.errors : null}
        loading={loading}
      />
    );
  }
}

const mapStateToProps = state => ({
    register: state.register,
  });

const mapDispatchToProps = dispatch => bindActionCreators({ registerUser }, dispatch);


export default connect(mapStateToProps, mapDispatchToProps)(RegisterView);
