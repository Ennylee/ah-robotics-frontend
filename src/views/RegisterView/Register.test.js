import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import RegisterView from './index';
import Adapter from 'enzyme-adapter-react-16';
import RegisterComponent from '../../components/Register';

Enzyme.configure({adapter: new Adapter()});

function setup() {
  // const props = {
  //   register: jest.fn()
  // };

  const registerSuccess = {
    register: {}
  };

  const wrapper = shallow(<RegisterComponent registerSuccess={registerSuccess}/>);

  return {
    // props,
    wrapper
  };
}


describe('Register View', () => {
  it('should render self and subcomponents', () => {
    const {wrapper} = setup();

    expect(wrapper.find('div').hasClass('login-form')).toBe(true);
    expect(wrapper.find('header')).toBeDefined();
    expect(wrapper.find('button')).toBeDefined();
  });
  it('app should render Registration page', () => {
    const wrapper = shallow(<RegisterView/>);
    expect(wrapper).toMatchSnapshot();
  });
});

