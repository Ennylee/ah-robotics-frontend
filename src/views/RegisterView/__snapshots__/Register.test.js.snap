// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register View app should render Registration page 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
